farm/modeling/prediction_head.py

```diff
@@ -1217,8 +1226,6 @@ def formatted_preds(self, logits=None, preds=None, baskets=None, **kwargs):
         # are prediction spans
         preds_d = self.aggregate_preds(preds, passage_start_t, ids, seq_2_start_t)
 
-        assert len(preds_d) == len(baskets)
```

Comment:
we should through a logger error here, since we need as many preds_d (on document level) as we have baskets.
If that is not the case it isnt necessarily breaking a cosuming app. Example: haystack retrieves 10 docs, one of those documents is malformatted so it cannot contain preds. Still we want to give back preds_d on all other 9 docs so the reader produces some answer